docs/reference/developer_guide/developing_custom_catalogs.md

```diff
@@ -93,3 +93,122 @@ The `SOperator` constructor method takes care of initializing the plug objects a
 The **execute** method takes care of defining the list of steps to be performed when the execution of the operator is issued inside Shift. In the example above, the execute method picks up the values of the two input plugs and multiplies them together. Finally, it stores the result in the output plug.
 
 Please refer to the Shift API documentation to know more about which plug types are present in Shift, and which additional methods the `SPlug` and `SOperator` classes offer.
+
+## Forcing Operators to Run on the Main Thread
+
+As mentioned in the [Execution Details](../../getting_started/basics/execute#executing-specific-nodes-on-the-main-thread) certain contexts require specific Python functions to be called from the main thread. This is the case when running `maya.cmds` or `nuke` calls.
```

Comment:
This is supposed to be the detailed explanation about running operators on the main thread. I don't think is necessary to reference a page that only gives a more swallow introduction
```suggestion
Certain contexts and applications require specific Python functions to be called from the main thread.  Ignoring this requirement can cause unexpected behaviours or crashes. This is the case, for instance, when running `maya.cmds` or `nuke` calls.
```